src/SeleniumLibrary/base/librarycomponent.py

```diff
@@ -56,3 +59,15 @@ def assert_page_not_contains(self, locator, tag=None, message=None,
                                  loglevel='INFO'):
         self.element_finder.assert_page_not_contains(locator, tag, message,
                                                      loglevel)
+
+    @property
+    def log_dir(self):
+        try:
+            logfile = BuiltIn().get_variable_value('${LOG FILE}')
+        except RobotNotRunningError:
+            logfile = os.getcwd()
+        if logfile != 'NONE':
+            logdir = os.path.dirname(logfile)
+        else:
+            logdir = BuiltIn().get_variable_value('${OUTPUTDIR}')
+        return logdir
```

Comment:
1. It took me a little time to make sure `RobotNotRunningError` cannot occur when cquerying `${OUTPUTDIR}`. I think the code should be reorganized a little.
2. There seems to be a logic bug when you actually get that `RobotNotRunningError`. The returned directory seems to be the parent directory of the cwd in that case.
3. `os.getcwd()` returns bytes on Python 2. This causes problems if the path contains non-ASCII characters. Should either use `os.getcwdu()` (which doesn't exist on Python 3) or just return `'.'`.

Perhaps the code could be something like this:
```python
try:
    logfile = BuiltIn().get_variable_value('${LOG FILE}')
    if logfile == 'NONE':
        return BuiltIn().get_variable_value('${OUTPUTDIR}')
    return os.path.dirname(logfile)
except RobotNotRunningError:
    return os.getcwd() if PY3 else os.getcwdu()
```

Reply:
Agreed and done.